File: app/src/main/java/com/chao/wanmvvm/model/BaseRepository.java
package com.chao.wanmvvm.model;

import com.chao.mvvm.base.AbsRepository;
import com.chao.mvvm.event.LiveBus;
import com.chao.mvvm.http.HttpHelper;
import com.chao.wanmvvm.network.ApiService;

/**
 * @author：tqzhang on 18/7/26 16:15
 */
public class BaseRepository extends AbsRepository {

    protected ApiService apiService;

    public BaseRepository() {
        if (null == apiService) {
            apiService = HttpHelper.getInstance().create(ApiService.class);
        }
    }


    protected void sendData(Object eventKey, Object t) {
        sendData(eventKey, null, t);
    }

    protected void sendData(Object eventKey, String tag, Object t) {
        LiveBus.getDefault().postEvent(eventKey, tag, t);
    }

    protected void showPageState(Object eventKey, String state) {
        sendData(eventKey, state);
    }

    protected void showPageState(Object eventKey, String tag, String state) {
        sendData(eventKey, tag, state);
    }


}

File: app/src/main/java/com/chao/wanmvvm/model/BaseRepository.kt
package com.chao.wanmvvm.model

import com.chao.mvvm.base.AbsRepository
import com.chao.mvvm.event.LiveBus
import com.chao.mvvm.http.RetrofitFactory
import com.chao.mvvm.stateview.StateConstants
import com.chao.wanmvvm.network.ApiService

/**
 * @author：tqzhang on 18/7/26 16:15
 */
open class BaseRepository : AbsRepository() {

    val apiService: ApiService by lazy {
        RetrofitFactory.getInstance().create(ApiService::class.java)
    }

    private fun sendData(eventKey: Any, t: Any) {
        sendData(eventKey, null, t)
    }

    private fun sendData(eventKey: Any, tag: String?, t: Any) {
        LiveBus.getDefault().postEvent(eventKey, tag, t)
    }

    fun showPageState(eventKey: Any, state: String) {
        sendData(eventKey, state)
    }

    private fun showPageState(eventKey: Any, tag: String, state: String) {
        sendData(eventKey, tag, state)
    }

    fun sendSuccessData(eventKeyState: Any,eventKey: Any,t: Any) {
        showPageState(eventKeyState, StateConstants.SUCCESS_STATE)
        sendData(eventKey,t)
    }

    fun sendSuccessData(eventKeyState: Any,eventKey: Any, tag: String, t: Any) {
        showPageState(eventKeyState,tag, StateConstants.SUCCESS_STATE)
        sendData(eventKey,tag,t)
    }

}
